CAIPs/caip-ipld_anchor_event.md

```diff
@@ -0,0 +1,129 @@
+
+### IPLD Based Digest
+
+In a general blockchain anchor implementation a message digest is published to the blockchain. There are many ways to create such a digest and implementors have to agree on construction, naming, resolution, etc. [IPLD CIDs](https://github.com/multiformats/cid) already serve this purpose by offering a standard self-describing format, cryptographic hashing and content addressing. 
+
+### IPLD Based Merkle Tree
+
+Typically users, applications, and services/protocols are interested in committing a set of data in a blockchain anchor. As it is can be inefficient to commit a single piece of data. Generally Merkle trees have been used to efficiently publish a blockchain anchor by publishing only the Merkle root over a set of inputs and allowing verification that a given piece of data was included in the set through a Merkle proof. 
+
+Again IPLD offers a natural and standard way to to encode a Merkle tree, with the added benefit of allowing anyone to easily resolve the entire tree, paths or subset of the tree when verifying a Merkle proof. We suggest encoding a IPLD Merkle tree as follows, but IPLD is self descriptive enough that verification is possible with any construction. 
+
+The set of data to be committed, or “leaves” of the Merkle tree are expected to be a set of IPLD CIDs. A binary Merkle tree is used, and a node value is constructed with two “leaves” or CIDs using a list. The data is encoded in IPLD with [dag-cbor](https://ipld.io/specs/codecs/dag-cbor/spec/), the resulting IPLD block results in another CID for this tree node. 
+
+IPLD Merke Tree Node Type
+
+```tsx
+type NodeList [Link]
+```
+
+Example 1
+
+```tsx
+[CID(bagcqcerav5zyrlelnnx24blbgob6stz6y4oiwenn3hodbmyf7bnnguh6gdeq), CID(bagcqcera4nnpzmwsk2bbww6n4bgq3cwne6gmhkouciam44nyjbindkxpz2eq)
+```
+
+Non leaf nodes are then recursively built in the same way using list pairs of CIDs until you have constructed the entire Merkle tree and a root Node/CID. 
+
+### IPLD Blockchain Anchor
+
+A blockchain anchor is made up of both a Merkle tree root CID and the corresponding reference to a blockchain transaction which includes the Merkle tree root. The blockchain transaction is referenced by a IPLD CID using the corresponding IPLD codec for the blockchain and the CAIP-2 chainID. There already exists IPLD codecs for two of the most commonly used blockchains for anchoring, Bitcoin and Ethereum. IPLD codec support may have to be added to support other chains. 
+
+```tsx
+type BlockchainAnchor struct {
+  root Link
+  chainID String 
+  txHash Link
+}
+```
+
+Example 2
+```tsx
+{
+  root: CID(bafyreiaxdhr5jbabn7enmbb5wpmnm3dwpncbqjcnoneoj22f3sii5vfg74)
+  chainID: "eip155:1"
+  txHash: CID(bagjqcgzanbud4sqdsywfp2mckuj57qsffsovgyjhh7sxebkqwr335hzy2zbq)
+}
+```
```

Comment:
We likely will need an additional property here to be able to distinguish between different transaction types. We basically need to know how to take the raw tx data and convert it to a CID. For example if it's a tx to a smart contract it will need a different way of decoding than if it's a regular tx.
```suggestion
```tsx
type BlockchainAnchor struct {
  root Link
  chainID String 
  txHash Link
  txType String
}
\```

Example 2
```tsx
{
  root: CID(bafyreiaxdhr5jbabn7enmbb5wpmnm3dwpncbqjcnoneoj22f3sii5vfg74)
  chainID: "eip155:1"
  txHash: CID(bagjqcgzanbud4sqdsywfp2mckuj57qsffsovgyjhh7sxebkqwr335hzy2zbq)
  txType: 'eth-raw'
}
\```
```

looks like the suggestion system doesn't really work over quote blocks...

Reply:
txType added, and ref to namespace for them